src/Dstarlite.js

```diff
@@ -0,0 +1,115 @@
+import Heap from './Heap';
+import Cell from './Cell';
+
+export default class Dstarlite {
+    constructor(start, goal, heuristic) {
+        this.Heap = new Heap();
+        this.km = 0;
+        this.rhs = new Map();
+        this.gscore = new Map();
+        this.heuristic = heuristic;
+        this.start = start;
+        this.goal = goal;
+    }
+
+    initialize() {
+        const grid = this.goal.grid.grid;
+        for (const row of grid) {
+            for (const column of row) {
+                this.gscore.set(column, Infinity);
+                this.rhs.set(column, Infinity);
+            }
+        }
+
+        this.rhs.set(this.goal, 0);
+        this.Heap.insert(this.goal, this.calculateKey(this.goal));
+    }
+
+    calculateKey(node) {
+        const min = Math.min(this.gscore.get(node), this.rhs.get(node));
+        return [
+            min + this.heuristic(this.start, node) + this.km,
+            min,
+        ];
+    }
+
+    compareKeys(key1, key2) {
+        if (key1[0] > key2[0] || (key1[0] === key2[0] && key1[1] > key2[1])) {
```

Comment:
can we simplify this?
i really rather
```js
if (key1[0] === key2[0]) {
  return key1[1] - key2[1];
}
return key1[0] - key2[0];
```
if it would confuse you, could just add documentation,
```js
// returns > 0 if key1 is lexicographically larger than key2
```

Reply:
🐇